engines/sci/engine/klists.cpp

```diff
@@ -824,6 +827,13 @@ reg_t kArraySetElements(EngineState *s, int argc, reg_t *argv) {
 }
 
 reg_t kArrayFree(EngineState *s, int argc, reg_t *argv) {
+	if (s->_segMan->getSegmentType(argv[0].getSegment()) == SEG_TYPE_SCRIPT) {
```

Comment:
No, this function is not properly named. It frees a single ELEMENT of an array, not the whole array:

    void SegManager::freeArray(reg_t addr) {
        // (...)
        ArrayTable &arrayTable = *(ArrayTable *)_heap[addr.getSegment()];
        // (...)
	arrayTable.freeEntry(addr.getOffset());
    }